src/SignalR/server/Core/src/IHubContext`T.cs

```diff
@@ -1,19 +1,25 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
+
 namespace Microsoft.AspNetCore.SignalR;
 
 /// <summary>
 /// A context abstraction for a hub.
 /// </summary>
-public interface IHubContext<THub, T>
+public interface IHubContext<THub, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>
```

Comment:
Did we want to put RequiresDynamicCode on this type like we did for the internal implementation?

Reply:
I would, but RequiresDynamicCode can only go on methods, constructors, and classes.

https://github.com/dotnet/aspnetcore/blob/0da8ea72b5434cbe8e1207d802f2270ca2f8ad4c/src/Shared/TrimmingAttributes.cs#L16-L17